force-app/main/default/classes/ServiceScheduleDomain.cls

```diff
@@ -36,8 +41,16 @@ public with sharing class ServiceScheduleDomain {
     }
 
     private void insertRelatedRecords(List<SObject> sObjects) {
-        validateCreateAccess(sObjects);
-        insert sObjects;
+        List<SObject> sObjectsNotYetInserted = new List<SObject>();
+        for (SObject sObj : sObjects) {
+            if (sObj.Id == null) {
```

Comment:
This feels a little off to me, we shouldn't be sending records with Ids to an insert method.

Reply:
Do you have a suggestion? We could filter out pre-existing records that have Ids outside of this method, or possibly re-work this as an upsert?